### PR TITLE
build: use full version string in setup.exe

### LINKED ISSUE #15546: gitian: Windows installer EXE filenames lack "rcN" suffix
gitian builds now add the "rcN" suffix to most filenames, but the Windows installer EXEs are lacking them.

### PR DESCRIPTION
**Pull request by maflcko**
Fixes: #15546

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK.
this would leave one file to be manually renamed
```
mv ${SOURCEDIR}/bitcoin-osx-signed.dmg ${DESTDIR}/bitcoin-${VERSIONTO}-osx.dmg
```

**Comment by laanwj** [comment]
One concern: how does this interact with the `gitian-win-signer.yml`? My memory isn't entirely clear on how it works, but would it need to be modified to expect the new name?

**Comment by MarcoFalke** [comment]
I think it signs anything that ends with `*-unsigned.exe`

File: share/setup.nsi.in [hunk 1]
```diff
@@ -59,12 +58,12 @@ CRCCheck on
 XPStyle on
 BrandingText " "
 ShowInstDetails show
-VIProductVersion ${VERSION}.@CLIENT_VERSION_BUILD@
+VIProductVersion @CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@.@CLIENT_VERSION_BUILD@
```
Comment (promag) [inline share/setup.nsi.in]: There are other places where this is done, like
https://github.com/bitcoin/bitcoin/blob/257f750cd986641afe04316ed0b22b646b56b60b/src/bitcoin-cli-res.rc#L5

Makes sense to define this in `configure.ac`?

**Review by promag — COMMENTED** [review]
utACK fa55104cb8e73c709e90fc5f81c9cb6a8a0713a6.

I agree it is better to use `PACKAGE_VERSION` and remove `define VERSION ...`

File: share/setup.nsi.in (hunk 1 above)
Comment (MarcoFalke) [reply share/setup.nsi.in]: Yeah, other PR. The goal of this one is really only to change the `OutFile`

**Pull request merged**